dropshot/src/handler.rs

```diff
@@ -326,7 +319,24 @@ impl HandlerError {
 
     pub(crate) fn into_response(self, request_id: &str) -> Response<Body> {
         match self {
-            Self::Handler { rsp, .. } => rsp,
+            // When the handler returns an error, we must add the request ID
+            // header to it now.
+            Self::Handler { mut rsp, .. } => {
+                match http::HeaderValue::from_str(request_id) {
+                    Ok(header) => {
+                        rsp.headers_mut()
+                            .insert(crate::HEADER_REQUEST_ID, header);
+                    }
+                    // This should never happen, but let's not panic in release
+                    // mode if it does.
+                    Err(e) if cfg!(debug_assertions) => {
+                        unreachable!("request ID {request_id:?} is not a valid HeaderValue: {e}");
+                    }
```

Comment:
Hmmm, the reason I didn't make this an `unwrap()` or similar is that elsewhere, when we set the request ID header when converting a  `HttpError` into a response, we do it through `http::response::Builder::header()`, which will cause `Builder::build()` to return an `Err` in the event of an invalid header value, rather than unwrapping it. See: https://docs.rs/http/latest/src/http/response.rs.html#611-624

So, perhaps what we really should do is instead return a 500, to be consistent with that behavior...but we already have a response explaining the original error, and it seemed sad to throw that away in favor of a 500 because we somehow generated a request ID that contained invalid characters.

Reply:
Oh, actually, I guess that `HttpError::into_response()` also `unwrap()`s the entire `Builder::build()` result, which includes this error: https://github.com/oxidecomputer/dropshot/blob/5377f2d6de478a0e23ef27f66531c8903e5f6b72/dropshot/src/error.rs#L303

...but on the other hand, there's a comment there saying we maybe shouldn't do that:

https://github.com/oxidecomputer/dropshot/blob/5377f2d6de478a0e23ef27f66531c8903e5f6b72/dropshot/src/error.rs#L279-L286

Reply:
I'll just make it panic unconditionally, I guess, although I prefer the match here rather than `unwrap`/`expect` as in this case, we know the specific header value that was invalid...